### ISSUE TITLE
FR: Search/Filter TOC

### ISSUE THREAD

**Issue by jonnyl2** [koreader/koreader#12921]
**Does your feature request involve difficulty completing a task? Please describe.**

Some books, especially ePubs consisting of a whole collection of an author's work, contain huge TOCs with sometimes dozens or even >100 pages. When looking for a certain chapter/novella/subject it is then not user-friendly to browse the entire TOC, and one has to resort to using Fulltext search and search the entire file to find the desired chapter in the file.

**Describe the solution you'd like**

An option to filter the TOC (e.g., via a new hamburger menu) by search terms. The TOC would then only list entries matching the search criteria entered.

**Describe alternatives you've considered**

n/a

**Additional context**

This feature would also be helpful for any kind of non-fiction book with a detailed TOC, to quickly list only subchapters dealing with a certain subject.

**Comment by jonnyl2** [comment]
No, those features just generate alternative TOCs. The feature I'm looking for would just be a temporary filter; it would reset the next time the TOC is opened. And it could be applied both to the main TOC or one of the alternative ones.

**Comment by mergen3107** [comment]
Good point! Alternatively for now, you can just search within the book, but only where pages with TOC are. 

We don't have a way to limit where search is being made though.

**Comment by Commodore64user** [comment]
> We don't have a way to limit where search is being made though.

I had made an FR about it #11454. ;)

**Comment by jonnyl2** [comment]
>I had made an FR about it  ;)

I remember that :) But it wasn't quite the same thing. It could have been used to search the "printed" TOC, but this one would strictly search the metadata TOC. Which is not necessarily the same (and is sometimes even more detailed, or even the only TOC available).

**Comment by jonnyl2** [comment]
Thank you, that's definitely useful in some cases, also for the TOC. But the TOCs I had in mind most for this FR are usually layered and so the items searched for are often buried in lower levels. Also it would be useful to find all results, especially when searching the TOC of manuals for example.

**Comment by jonnyl2** [comment]
For reference; something like this is what I had in mind (a filtered view with only the entries matching the search term):

![Image](https://github.com/user-attachments/assets/29a92f12-8675-4951-b844-b1cbd5054d93)
